src/metabase/driver/sql/query_processor.clj

```diff
@@ -109,12 +121,17 @@
 
 ;; default implementation for `:default` bucketing returns expression as-is
 (defmethod date [:sql :default] [_ _ expr] expr)
-
 ;; We have to roll our own to account for arbitrary start of week
-(defmethod date [:sql :week-of-year]
-  [driver _ expr]
+
+(defmethod date [:sql :second-of-minute] [_driver _ expr] (hx/second expr))
+(defmethod date [:sql :minute-of-hour]   [_driver _ expr] (hx/minute expr))
+(defmethod date [:sql :hour-of-day]      [_driver _ expr] (hx/hour expr))
+(defmethod date [:sql :week-of-year]     [driver _ expr]
```

Comment:
I wonder why we don't want to have week-of-year extractor.

Reply:
I'll check with the product team. if we do I'll add support for it in a follow up.